spec.html

```diff
@@ -11059,6 +11060,44 @@ <h1>Examples of Automatic Semicolon Insertion</h1>
+      <h1>Automatic Semicolon Insertion Hazards in Statement Lists</h1>
+      <p>In a |StatementList|, many |StatementListItem|s end in semicolons, which may be omitted using automatic semicolon insertion. As a consequence of the rules above, at the end of a line ending an expression, a semicolon is required if the following line begins with any of the following:
+      <ul>
+        <li><strong>An opening parenthesis (<code>(</code>)</strong>. Without a semicolon, the two lines together are treated as a |CallExpression|.</li>
+        <li><strong>An opening square bracket (<code>[</code>)</strong>. Without a semicolon, the two lines together are treated as property access, rather than an |ArrayLiteral| or |ArrayAssignmentPattern|.</li>
+        <li><strong>A template string (<code>`</code>)</strong>. Without a semicolon, the two lines together are interpreted as a tagged Template (<emu-xref href="#sec-tagged-templates"></emu-xref>), with the previous expression as the |MemberExpression|.</li>
+        <li><strong>Unary <code>+</code> or <code>-</code></strong>. Without a semicolon, the two lines together are interpreted as a usage of the corresponding binary operator.</li>
+        <li><strong>A RegExp literal</strong>. Without a semicolon, the two lines together may be parsed instead as the `/` |MultiplicativeOperator|, for example if the RegExp has flags.</li>
+      </ul>
+    </emu-clause>
+  </emu-clause>
+  <emu-clause id="sec-no-lineterminator-here-automatic-semicolon-insertion">
+    <h1>Automatic Semicolon Insertion and &ldquo;[no |LineTerminator| here]&rdquo;</h1>
+    <em>This section is non-normative.</em>
+    <p>ECMAScript contains grammar productions which include &ldquo;[no |LineTerminator| here]&rdquo;. These productions are sometimes a means to have optional operands in the grammar. Introducing a |LineTerminator| in these locations would change the grammar production of a source text by using the grammar production without the optional operand. These locations are not considered hazardous because additive changes would need to alter source text within a grammar production that is subject to automatic semicolon insertion in order to change the grammar production.</p>
+
+    <p>The rest of this section describes a number of productions using &ldquo;[no |LineTerminator| here]&rdquo; in this version of ECMAScript.</p>
+    <emu-clause id="sec-no-lineterminator-here-automatic-semicolon-insertion-list">
+      <h1>List of Grammar Productions with Optional Operands and &ldquo;[no |LineTerminator| here]&rdquo;</h1>
+      <ul>
+        <li>|UpdateExpression|.</li>
+        <li>|ContinueStatement|.</li>
+        <li>|BreakStatement|.</li>
+        <li>|ReturnStatement|.</li>
+        <li>|YieldExpression|.</li>
+        <li>Async Function Definitions (<emu-xref href="#sec-async-function-definitions"></emu-xref>) with relation to Function Definitions (<emu-xref href="#sec-function-definitions"></emu-xref>) </li>
```

Comment:
s/definition/declaration/. Also, async method declarations.

Reply:
@littledan linking to sections w/ productions not individual grammar productions on this on

@claudepache `ThrowStatement` and `ArrowFunction` are not a source of this issue due to the operand not being optional.